File: ShoeStore/ShoeStore.Domain/Entities/ReceiptDetail.cs
namespace ShoeStore.Domain.Entities
{
    public class ReceiptDetail
    {
        public long Id { get; set; }
        public long ReceiptId { get; set; }
        public int ProductId { get; set; }
        public int QuantityOrdered { get; set; }
        public int? ReceivedQuantity { get; set; }
        public decimal UnitPrice { get; set; }

        public Receipt Receipt { get; set; } = null!;
        public Product Product { get; set; } = null!;
    }
}

File: shoestorefe/lib/domain/entities/receipt_detail.dart
class ReceiptDetail {
  final int id;
  final int productId;
  final String? productName;
  final String? sku;
  final int quantityOrdered;
  final int? receivedQuantity;
  final double unitPrice;

  ReceiptDetail({
    required this.id,
    required this.productId,
    this.productName,
    this.sku,
    required this.quantityOrdered,
    this.receivedQuantity,
    required this.unitPrice,
  });
}
